lib/api/src/js/externals/memory_view.rs

```diff
@@ -0,0 +1,186 @@
+use crate::js::store::AsStoreRef;
+use crate::js::MemoryAccessError;
+use std::convert::TryInto;
+use std::marker::PhantomData;
+use std::mem::MaybeUninit;
+use std::slice;
+use tracing::warn;
+
+use wasmer_types::{Bytes, Pages};
+
+use super::Memory;
+use super::memory::MemoryBuffer;
+
+/// A WebAssembly `memory` view.
+///
+/// A memory view is used to read and write to the linear memory.
+///
+/// After a memory is grown a view must not be used anymore. Views are
+/// created using the Memory.grow() method.
+#[derive(Debug)]
+pub struct MemoryView<'a> {
+    view: js_sys::Uint8Array,
+    size: u64,
+    marker: PhantomData<&'a Memory>,
+}
+
+impl<'a> MemoryView<'a>
+{
+    pub(crate) fn new(memory: &Memory, store: &impl AsStoreRef) -> Self {
+        let buffer = memory
+            .handle
+            .get(store.as_store_ref().objects())
+            .memory
+            .buffer();
+
+        let size = js_sys::Reflect::get(
+                &buffer,
+                &"byteLength".into(),
+            )
+            .unwrap()
+            .as_f64()
+            .unwrap() as u64;
+
+        let view = js_sys::Uint8Array::new(
+            &buffer,
+        );
+
+        Self {
+            view,
+            size,
+            marker: PhantomData,
+        }
+    }
+
+    /// Returns the pointer to the raw bytes of the `Memory`.
+    #[doc(hidden)]
+    pub fn data_ptr(&self) -> *mut u8 {
+        unimplemented!("direct data pointer access is not possible in JavaScript");
+    }
+
+    /// Returns the size (in bytes) of the `Memory`.
+    pub fn data_size(&self) -> u64 {
+        self.size
+    }
+
+    /// Returns the size (in [`Pages`]) of the `Memory`.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # use wasmer::{Memory, MemoryType, Pages, Store, Type, Value};
+    /// # let mut store = Store::default();
+    /// #
+    /// let m = Memory::new(&store, MemoryType::new(1, None, false)).unwrap();
+    ///
+    /// assert_eq!(m.size(), Pages(1));
+    /// ```
+    pub fn size(&self) -> Pages {
+        Bytes(self.size as usize).try_into().unwrap()
+    }
+
+    /// Used by tests
+    #[doc(hidden)]
+    pub fn uint8view(&self) -> js_sys::Uint8Array {
+        self.view.clone()
+    }
```

Comment:
Now that we have the `MemoryView` I believe we can remove all usages of this function `uint8view`, and make them use the read, write instead. That way the API is fully uniform between js/sys

Reply:
done